### PR TITLE
Use shared error components

### PR DESCRIPTION
**Pull request by tothszabi**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Does not require a release.

### Context

This PR does not add new functionality at all. It just replaces the already existing error formatter in this step with a shared one which has exactly the same feature set.

### CODE REVIEW

File: dependencies.go [hunk 1]
```diff
@@ -1,9 +1,8 @@
 package main
 
 import (
-	"os"
-
 	"github.com/bitrise-io/go-utils/v2/command"
+	"os"
```
Comment (godrei) [inline dependencies.go]: NIT: import order

File: main.go [hunk 2]
```diff
@@ -38,14 +39,14 @@ func run() ExitCode {
...
-		buildStep.logger.Errorf(formattedError(fmt.Errorf("Failed to install Step dependencies: %w", err)))
+		buildStep.logger.Errorf(errorutil.FormattedError(fmt.Errorf("Failed to install Step dependencies: %w", err)))
 		return Failure
 	}
 
 	runOpts := createRunOptions(config)
 	if err := buildStep.Run(runOpts); err != nil {
 		buildStep.logger.Println()
-		logger.Errorf(formattedError(fmt.Errorf("Failed to execute Step main logic: %w", err)))
+		logger.Errorf(errorutil.FormattedError(fmt.Errorf("Failed to execute Step main logic: %w", err)))
```
Comment (godrei) [inline main.go]: Similarly to this: https://github.com/bitrise-steplib/steps-xcode-test/pull/219#discussion_r1044560133

File: dependencies.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func (f FastlaneRunner) InstallDependencies(opts EnsureDependenciesOpts) error {
 			f.logger.Println()
 
 			if err := cmd.Run(); err != nil {
-				return f.wrapCommandError(cmd, err)
```
Comment (vshah23) [inline dependencies.go:36]: This method was adding some text as part of the human readable messaging. Are we removing that to go back to the old style?

Comment (vshah23) [reply dependencies.go:36]: Nevermind. I see how this is working now.

**Review by vshah23 — APPROVED** [review]

**Pull request merged**